docs/acceptance-tests.md

```diff
@@ -0,0 +1,298 @@
+# Acceptance tests
+
+This document describes one possible approach for writing acceptance tests for Conduit
+connectors. We discovered the need to revisit this when integrating the OpenCDC record
+into the SDK, since the new record format fully enables the implementation of mutating
+connectors.
+
+The main challenge is that 3rd party systems have different capabilities, it is not
+always possible to write a fully fledged mutable connector. Therefore the acceptance
+tests need to be flexible and adjust their expectations based on what the 3rd party
+system is capable of. At first it's probably simplest to let the developer specify
+which category the connector falls into and which tests should be run. In the future
+it would be ideal if we can detect the category automatically.
+
+We also need to confirm that the way we randomly generate records and how we assert
+them makes sense for all connectors.
+
+This document should serve as a basis for discussion, it does not explain the current
+implementation, nor should it be taken as the one single truth for how we should
+approach the writing of acceptance tests.
+
+## Source
+
+```mermaid
+flowchart TD;
+    %% Source
+    srcD1{"Does the<br>source support<br>snapshots?"};
+    srcD2{"Can the source<br>detect mutations?<br>(updates, deletes)"};
+    srcSignaling["Signaling source"];
+    srcAppending["Appending source"];
+    srcMutating["Mutating source"];
+
+    style srcD1 fill:#eee,stroke:black
+    style srcD2 fill:#eee,stroke:black
+    style srcSignaling fill:Moccasin,stroke-width:0
+    style srcAppending fill:PaleGoldenrod,stroke-width:0
+    style srcMutating fill:YellowGreen,stroke-width:0
+
+    srcD1 -- No --> srcSignaling;
+    srcD1 -- Yes ---> srcD2;
+    srcD2 -- No --> srcAppending;
+    srcD2 -- Yes --> srcMutating;
+```
+
+All source tests are executed with structured records. The reason is that not all destinations are able to consume unstructured raw data, while structured data should be supported by any destination connector. We should give the user control over field names in the structured record in case the connector expects specific field names that match the target structure.
+
+| Test                                                 | ![Signaling source](https://img.shields.io/badge/-Signaling%20source-yellow?style=flat-square) | ![Appending source](https://img.shields.io/badge/-Appending%20source-yellowgreen?style=flat-square) | ![Mutating source](https://img.shields.io/badge/-Mutating%20source-green?style=flat-square) |
```

Comment:
maybe add some info about what this table presents. and that you'll get into more details in the doc

Reply:
I added a sentence describing the table.